src/VisualStudio/Core/Def/Implementation/ProjectSystem/AnalyzerConfigDocumentAsSolutionItemHandler.cs

```diff
@@ -93,10 +108,14 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                     return;
                 }
 
-                var infoBarService = _workspace.Services.GetRequiredService<IInfoBarService>();
-                infoBarService.ShowInfoBarInGlobalView(
-                    ServicesVSResources.A_new_editorconfig_file_was_detected_at_the_root_of_your_solution_Would_you_like_to_make_it_a_solution_item,
-                    GetInfoBarUIItems().ToArray());
+                if (!_infoBarShownForCurrentSolution)
+                {
+                    _infoBarShownForCurrentSolution = true;
```

Comment:
Any reason not to just set this once we've kicked off the task? Then it avoids having to re-check it a second time which seems a bit easier.